tests/test_lime_survey.py

```diff
@@ -208,6 +209,103 @@ def test_lime_system_info_dtypes(self):
             True,
         )
 
+class TestLimeSurveyGetResponse(BaseTestLimeSurvey2021Case):
+    """Test LimeSurvey get response"""
+
+    def setUp(self) -> None:
+        """Read responses before each test."""
+        self.survey.read_responses(responses_file=self.responses_file)
+
+    def test_get_response(self):
+        """Test basic get response functionality for valid and invalid input"""
+        try:
+            questions = [
+                'A1', 'A3', 'A4', 'A5', 'A10',
+                'A14', 'B6', 'D2', 'E5a', 'E9'
+            ]
+            for question in questions:
+                self.survey.get_responses(question, labels=False)
+                self.survey.get_responses(question, labels=True)
+        except:
+            self.fail('No assertion expected for valid input.')
+
+        # assume exception raised after invalid question name
+        self.assertRaises(ValueError, self.survey.get_responses, 'X2', labels=False)
+        self.assertRaises(ValueError, self.survey.get_responses, 'X2', labels=True)
+
+        # assume exception is thrown after A2 missing in responses
+        self.assertRaises(KeyError, self.survey.get_responses, 'A2', labels=False)
+        self.assertRaises(KeyError, self.survey.get_responses, 'A2', labels=True)
+
+    def test_get_response_single_choice(self):
+        """Test get response for single choice question type"""
+        expected_response = [
+            'A1', 'A1', 'A3', 'A2', 'nan',
+            'A1', 'A1', 'A1', 'A1', 'A1'
+        ]
+        response = self.survey.get_responses(self.single_choice_column, labels=False)
+        np.testing.assert_array_equal(
+            expected_response,
+            response.values.flatten().astype(str)[:10]
+        )
+
+        expected_response = [
+            'Woman', 'Woman', 'Man', "I don't want to answer this question",
+            'nan', 'Woman', 'Woman', 'Woman', 'Woman', 'Woman'
+        ]
+        response = self.survey.get_responses(self.single_choice_column, labels=True)
+        np.testing.assert_array_equal(
+            expected_response,
+            response.values.flatten().astype(str)[:10]
+        )
+
+    def test_get_response_multiple_choice(self):
+        """Test get response for multiple choice question type"""
+        expected_response = [
+            False, False, False, False, False,
+            False, False, False, False, False
+        ]
+        response = self.survey.get_responses(self.multiple_choice_column, labels=False)
+        np.testing.assert_array_equal(expected_response, response.values.flatten()[:10])
+        response = self.survey.get_responses(self.multiple_choice_column, labels=True)
+        np.testing.assert_array_equal(expected_response, response.values.flatten()[:10])
+
```

Comment:
Let's add column names to the test. Please, also check the contingent (i.e. "Other") question

Reply:
I can add this but not before friday I think.